integration_tests/src/main/scala/com/nvidia/spark/rapids/tests/common/BenchUtils.scala

```diff
@@ -127,14 +129,16 @@ object BenchUtils {
     val queryStartTime = Instant.now()
 
     val queryPlansWithMetrics = new ListBuffer[SparkPlanNode]()
+    val stageMetrics = new ListBuffer[StageMetrics]()
 
     var df: DataFrame = null
     val queryTimes = new ListBuffer[Long]()
     for (i <- 0 until iterations) {
 
-      // capture spark plan metrics on the final run
+      // capture spark metrics on the final run
       if (i+1 == iterations) {
         spark.listenerManager.register(new BenchmarkListener(queryPlansWithMetrics))
+        spark.sparkContext.addSparkListener(new BenchSparkListener(stageMetrics))
```

Comment:
we have BenchmarkListener and BenchSparkListener, both of which I'm not sure what are from just the names.  Perhaps we should give the more meaningful names.   I get that they might be used for more then just stage metrics for instance, but would be nice to differentiate them somehow or use a common one.